### PR TITLE
Split `Organization.LimitCollectionCreationDeletion` into two separate business rules

### PR DESCRIPTION
**Pull request by addisonbeck**
### 📚 Stacked PRs

1. `server`: https://github.com/bitwarden/server/pull/4709
1. `server`: https://github.com/bitwarden/server/pull/4730
1. `clients`: https://github.com/bitwarden/clients/pull/11223
    ⬆️ __YOU ARE HERE__
1. `server`: https://github.com/bitwarden/server/pull/4808
1. `server`: https://github.com/bitwarden/server/pull/4809
1. `clients`: https://github.com/bitwarden/clients/pull/11258
1. `server`: https://github.com/bitwarden/server/pull/4810

### 🤳 The Bigger Picture 

There is an option in the Admin Console that removes collection creation and
deletion controls from all organization members that are not in the Owner and
Admin roles. This PR stack splits this conjoined option into two separate
controls: one to block `create` operations and one to block `delete`
operations. This is a part of an effort to make these settings better align
with what customers expect it to do.

### 👉 This Pull Request 👍

In the previous PR we implemented the new functionality requested from this
effort in `server`. On this PR we apply those new rules to `clients`. This
involves adding the feature toggle to `clients` and implementing two new
boolean settings in the admin console behind that feature toggle.

This PR, its predecessor in `server`, and the next PR in the stack encompass
all of the functional changes to the system implemented by this PR stack. The
first pull request and the last two in the stack are for maintenance to
support evolutionary database design and feature toggling.

### 🧮 Side Effects

- Because this work included removing `LimitCollectionCreationDeletion` and
  `AllowAdminsAccessToAllCollections` from the license file used for self
  hosted: the new controls are not disabled for self hosted organizations.

## 📸 Screenshots

Several recorded demos can be found in Jira

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-10863]: https://bitwarden.atlassian.net/browse/PM-10863?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/web/src/app/admin-console/organizations/settings/account.component.ts [hunk 1]
```diff
@@ -61,6 +66,15 @@ export class AccountComponent implements OnInit, OnDestroy {
     }),
   });
 
+  protected collectionManagementFormGroup_VNext = this.formBuilder.group({
```
Comment (eliykat) [inline apps/web/src/app/admin-console/organizations/settings/account.component.ts:69]: I can see the `vNext` formgroup is used in the TS code, but it doesn't appear to be referenced in the template.

File: apps/web/src/app/admin-console/organizations/settings/account.component.ts [hunk 2]
```diff
@@ -78,11 +92,17 @@ export class AccountComponent implements OnInit, OnDestroy {
...
+    private configService: ConfigService,
   ) {}
 
   async ngOnInit() {
     this.selfHosted = this.platformUtilsService.isSelfHost();
 
+    this.configService
+      .getFeatureFlag$(FeatureFlag.LimitCollectionCreationDeletionSplit)
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((x) => (this.limitCollectionCreationDeletionSplitFeatureFlagIsEnabled = x));
```
Comment (eliykat) [inline apps/web/src/app/admin-console/organizations/settings/account.component.ts:104]: I recommend against subscribing, then you can just do a 1-liner: `this.limitCollectionCreationDeletionSplitFeatureFlagIsEnabled = this.configService.getFeatureFlag$(FeatureFlag.LimitCollectionCreationDeletionSplit);`.

Although I admit using `await firstValueFrom` is a bit awkward, that may come down to personal preference.

**Review by eliykat — CHANGES_REQUESTED** [review]

File: apps/web/src/app/admin-console/organizations/settings/account.component.ts (hunk 1 above)
Comment (addisonbeck) [reply apps/web/src/app/admin-console/organizations/settings/account.component.ts:69]: Thanks for the catch. Addressed!

__Commit:__ [Reference feature toggle in template](https://github.com/bitwarden/clients/pull/11223/commits/c712b074bc34cec7e52b5e64313b7f93ffb7f27b)

File: apps/web/src/app/admin-console/organizations/settings/account.component.ts (hunk 2 above)
Comment (addisonbeck) [reply apps/web/src/app/admin-console/organizations/settings/account.component.ts:104]: I do tend to use observables over promises as a principle. If you feel strongly about this let me know, but I don't think this matters too much for such ephemeral code. Both are definitely readable and people are used to observable data flows in the project.

Comment (eliykat) [reply apps/web/src/app/admin-console/organizations/settings/account.component.ts:104]: Not at all, I agree completely about it being ephemeral code.

**Review by eliykat — APPROVED** [review]

**Review by JaredSnider-Bitwarden — APPROVED** [review]
Auth change looks good

**Review by eliykat — APPROVED** [review]

**Pull request merged**